fuel-database/src/error.rs

```diff
@@ -0,0 +1,132 @@
+use crate::InterpreterError;
+use std::io::ErrorKind;
+use thiserror::Error;
+
+// TODO: Use either `KvStoreError` or `Error`
```

Comment:
As a library I believe the correct way would be hierarchical errors, so external facing functions all return a generic `DatabaseError`, which is itself a wrapper for various errors like `IoError` (which may occur writing data) or `InitializationError` (an error wrapper for all initialization related errors) 

Looks like this has been done to some degree, but would be good to finalize an external error enum before merging

Reply:
I agree, but this PR is simple movement stuff from one place to another. The issue with errors is mentioned in the `follow-ups` section in the description=)